aea/aea_builder.py

```diff
@@ -559,10 +559,22 @@ def _process_connection_ids(
             ]
         else:
             selected_connections_ids = [
-                k.public_id for k in self._package_dependency_manager.connections.keys()
+                component_id.public_id
+                for component_id in self._package_dependency_manager.connections.keys()
             ]
 
-        return selected_connections_ids
+        # sort default id to be first
```

Comment:
@marcofavorito note, you forgot to ensure that the default connection is at index 0. I added this now.

Reply:
so that means the default connection is always run, even if I specify the `--connections` option?

Reply:
Yes, it seems problematic to not require the `default_connection` to be present even when the flag is used and I don't see what's the benefit in supporting such a "misconfiguration"